scanners/reactnative-expo/reactnative-expo.go

```diff
@@ -0,0 +1,674 @@
+
+			npmCmd := command.New("npm", "install")
+			npmCmd.SetDir(projectDir)
+			if out, err := npmCmd.RunAndReturnTrimmedCombinedOutput(); err != nil {
+				return models.OptionModel{}, warnings, fmt.Errorf("failed to npm install react-native in: %s\noutput: %s\nerror: %s", projectDir, out, err)
+			}
+
+			options, warns, err := androidScanner.Options()
+			warnings = append(warnings, warns...)
+			if err != nil {
+				return models.OptionModel{}, warnings, err
+			}
+
+			androidOptions = &options
+			scanner.androidScanner = androidScanner
+		}
+	}
+
+	// ios options
+	var iosOptions *models.OptionModel
+	iosDir := filepath.Join(projectDir, "ios")
+	if exist, err := pathutil.IsDirExists(iosDir); err != nil {
+		return models.OptionModel{}, warnings, err
+	} else if exist {
+		iosScanner := ios.NewScanner()
+
+		if detected, err := iosScanner.DetectPlatform(scanner.searchDir); err != nil {
+			return models.OptionModel{}, warnings, err
+		} else if detected {
+			options, warns, err := iosScanner.Options()
+			warnings = append(warnings, warns...)
+			if err != nil {
+				return models.OptionModel{}, warnings, err
+			}
+
+			iosOptions = &options
+			scanner.iosScanner = iosScanner
+		}
+	}
+
+	if androidOptions == nil && iosOptions == nil {
+		return models.OptionModel{}, warnings, errors.New("no ios nor android project detected")
+	}
+	// ---
+
+	if androidOptions != nil {
+		if iosOptions == nil {
+			// we only found an android project
+			// we need to update the config names
+			lastChilds := androidOptions.LastChilds()
+			for _, child := range lastChilds {
+				for _, child := range child.ChildOptionMap {
+					if child.Config == "" {
+						return models.OptionModel{}, warnings, fmt.Errorf("no config for option: %s", child.String())
+					}
+
+					configName := configName(true, false, hasNPMTest)
+					child.Config = configName
+				}
+			}
+		} else {
+			// we have both ios and android projects
+			// we need to remove the android option's config names,
+			// since ios options will hold them
+			androidOptions.RemoveConfigs()
+		}
+
+		rootOption = *androidOptions
+	}
+
+	if iosOptions != nil {
+		lastChilds := iosOptions.LastChilds()
+		for _, child := range lastChilds {
+			for _, child := range child.ChildOptionMap {
+				if child.Config == "" {
+					return models.OptionModel{}, warnings, fmt.Errorf("no config for option: %s", child.String())
+				}
+
+				configName := configName(scanner.androidScanner != nil, true, hasNPMTest)
+				child.Config = configName
+			}
+		}
+
+		if androidOptions == nil {
+			// we only found an ios project
+			rootOption = *iosOptions
+		} else {
+			// we have both ios and android projects
+			// we attach ios options to the android options
+			rootOption.AttachToLastChilds(iosOptions)
+		}
+
+	}
+
+	return rootOption, warnings, nil
+}
+
+// DefaultOptions ...
+func (Scanner) DefaultOptions() models.OptionModel {
+	gradleFileOption := models.NewOption(gradleFileInputTitle, gradleFileInputEnvKey)
```

Comment:
could not we use the ios and android scanner's default options?

Reply:
I think it would be better if we duplicate this one because it could be difficult to modify the Android or iOS scanner if we use them everywhere.

Reply:
that is true, but we want to use the same options here

Reply:
other than this you add this option, but is not used in any steo